include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -990,6 +988,16 @@ struct OrtApi {
    * This is a no-copy method whose pointer is only valid until the backing OrtValue is free'd.
    */
   ORT_API2_STATUS(TensorAt, _Inout_ OrtValue* value, size_t* location_values, size_t location_values_count, _Outptr_ void** out);
+
+  /*
+  * Creates a custom environment with global threadpools and logger that will be shared across sessions.
+  * Use this in conjunction with DisablePerSessionThreads API or else the session will use
+  * its own thread pools.
+  * 
+  * \param out should be freed by `OrtReleaseEnv` after use
+  */
+  ORT_API2_STATUS(CreateCustomEnv, OrtLoggingFunction logging_function, _In_opt_ void* logger_param, OrtLoggingLevel default_logging_level,
```

Comment:
I think adding a SetCustomLogger(_In_ const OrtEnv* env) function might be better. It gives us the flexibility to add more CreateEnv* variants later without creating the burden to accept a custom logger in those variants.

Reply:
Not sure I understand. The argument variants already exist - they unfortunately did not allow for the combination of **all** arguments at the same time. A `SetCustomLogger(...)` function would be nice but should that be done in another PR?

Although verbose, I could rename the function to `CreateEnvWithCustomLoggerAndGlobalThreadPools(...)` to coincide with other functions naming style.

Reply:
@pranavsharma Sorry for the delay (been on vacation). Renamed the function as we discussed before and updated the branch from master. Should be good to go.